src/control/cmd/daos_agent/fabric.go

```diff
@@ -92,9 +92,10 @@ type NUMAFabric struct {
 
 	numaMap NUMAFabricMap
 
-	currentNumaDevIdx map[int]int // current device idx to use on each NUMA node
-	currentNUMANode   int         // current NUMA node to search
-	ignoreIfaces      common.StringSet
+	currentNumaDevIdx  map[int]int      // current device idx to use on each NUMA node
+	currentNUMANode    int              // current NUMA node to search
+	ifaceFilterSet     common.StringSet // set of interface names for filtering
+	ifaceFilterExclude bool             // true: exclude filtered interfaces, false: include only filtered interfaces
```

Comment:
bool is not the best design here (false meaning include? when the size of the set above is not 0?)

consider instead defining a new type (ifaceFilter) with mode (include/exclude as const/enum? and then not needing to check the size of the set but check if filter is nil or not.

Reply:
Hmm. I didn't want to overengineer this as it's all internal to the agent, but I can take another look at it. I didn't really like this either, tbh.

Reply:
An alternative could be to pass in the desired filter function as a parameter directly.